rose/server/score.py

```diff
@@ -14,12 +14,19 @@ def process(players, track):
     # status, used for resolving collisions.
 
     for player in players.itervalues():
+        player_start_point = player.lane * config.cells_per_player
         if player.action == actions.LEFT:
-            if player.x > 0:
+            # the left player should be start on x=0, and right in the middle
+            if player.x > player_start_point:
                 player.x -= 1
+	    else:
+		player.score -= config.move_backward
+
         elif player.action == actions.RIGHT:
-            if player.x < config.matrix_width - 1:
+            if player.x < player_start_point + config.cells_per_player - 1:
                 player.x += 1
+	    else:
+		player.score -= config.move_backward
 
```

Comment:
Please separate this to another patch, this is not related to replacing life with score.

Are you trying to prevent player from moving to the other lane? I don't think we want this behavior.